api/host.go

```diff
@@ -121,38 +119,59 @@ func ModifyHostStatus(queue amboy.Queue, h *host.Host, newStatus string, notes s
 		currentStatus == evergreen.HostQuarantined &&
 		utility.StringSliceContains([]string{evergreen.HostRunning, evergreen.HostProvisioning}, newStatus)
 	if unquarantinedAndNeedsReprovision {
-		if err := h.SetNeedsReprovisionToNew(u.Username()); err != nil {
+		if _, err = GetReprovisionToNewCallback(ctx, env, u.Username())(h); err != nil {
```

Comment:
here too: don't we need to wait for the agent monitor to stop?

Reply:
Changed it to check the agent monitor before enqueue.